Content.Server/Fluids/EntitySystems/EvaporationSystem.cs

```diff
@@ -0,0 +1,73 @@
+﻿using Content.Server.Fluids.Components;
+using Content.Shared.Chemistry.EntitySystems;
+using Content.Shared.Chemistry.Reagent;
+using JetBrains.Annotations;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Utility;
+
+namespace Content.Server.Fluids.EntitySystems
+{
+    [UsedImplicitly]
+    public sealed class EvaporationSystem : EntitySystem
+    {
+        [Dependency] private readonly SolutionContainerSystem _solutionContainerSystem = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<EvaporationComponent, ComponentInit>(OnComponentInit);
+        }
+
+        private void OnComponentInit(EntityUid uid, EvaporationComponent component, ComponentInit args)
+        {
+            _solutionContainerSystem.EnsureSolution(uid, component.SolutionName);
+        }
+
+        public override void Update(float frameTime)
+        {
+            base.Update(frameTime);
+            var queueDelete = new RemQueue<EvaporationComponent>();
+            foreach (var evaporationComponent in ComponentManager.EntityQuery<EvaporationComponent>())
+            {
+                UpdateEvaporation(evaporationComponent, frameTime, ref queueDelete);
+            }
+
+            foreach (var evaporationComponent in queueDelete)
+            {
+                ComponentManager.RemoveComponent(evaporationComponent.Owner.Uid, evaporationComponent);
+            }
+        }
+
+        private void UpdateEvaporation(EvaporationComponent evaporationComponent, float frameTime,
+            ref RemQueue<EvaporationComponent> queueDelete)
+        {
+            var uid = evaporationComponent.Owner.Uid;
+            evaporationComponent.Accumulator += frameTime;
+
+            if (!_solutionContainerSystem.TryGetSolution(uid, evaporationComponent.SolutionName, out var solution))
+                return;
+
+            if (evaporationComponent.Accumulator < evaporationComponent.EvaporateTime)
+                return;
```

Comment:
Technically it's possible the component lives there forever if there's no solution; in that situation it should just delete the entity.